### DISCUSSION TITLE
How to move cursor to message popup?

### DISCUSSION THREAD

**Discussion by firubat** [LazyVim/LazyVim#3113] [Q&A]
I'm trying to tweak the colorscheme using `:Inspect!` . It brings up the information in a message popup, that disappears after a few seconds. I want to move the cursor to that message so I can yank the highlight group. How do I do that? 
I'm not sure what plugin controls the message popup, and seems it's not considered a buffer or window that I can move to with the respective keymaps...
I also tried using the flash search function but the message is not included.

**Comment by dpetka2001** [comment]
Just use `:messages` or `:Noice all`.

**Reply by firubat** [reply]
That could be a workaround if I can get it to display the last message first, is there such an option? 
Or I can make a keymap that will go directly to the end of the list.

But still it's weird I can't get directly to the popup. If I use the mouse I can click the popup to go to it, but surely that can't be the only way...

**Reply by dpetka2001** [reply]
You can use `<C-w>w` to switch to other windows (default Neovim behavior).

**Reply by firubat** [reply]
OK, so seems that the message is indeed just a window (contrary to what I wrote before) but it seems it is set to be the <s>LAST</s> window in the `<C-w>w` list, so if you have more than one window, using just `<C-w>w` you'd have go through all of them. 

<s>This can be overcome by using a large counter, e.g. `99<C-w>w` will most likely do the trick (can't think of anyone using that much windows...)
Would be a handy keybind to add ("move cursor to last window"), don't you think?</s>

EDIT: I was wrong in my assumption (although it worked at a certain point), the above keybind moves me to the number list...

**Reply by dpetka2001** [reply]
The correct way to view messages is either by `:messages` or `:Noice history`. You can also use `:Telescope notify` and if you select the message you want it will open in a float window.

**Comment by firubat** [comment]
Seems that the message popup is a window which is set to be the last window as understood by the function `winnr()` (see [here](https://neovim.io/doc/user/builtin.html#winnr()) ). So a possible keymap would be:
```
vim.keymap.set(
  "n",
  "<C-w>m",
  ":exe winnr('$') .. 'wincmd w'<CR>",
  { silent = true, desc = "Go to the message window (or last window)" }
)
```
